commit_test.go

```diff
@@ -455,3 +455,20 @@ func TestCommitEqualReturnsTrueWhenBothCommitsAreNil(t *testing.T) {
 
 	assert.True(t, c1.Equal(c2))
 }
+
+func TestBadCommit(t *testing.T) {
+	cc := `tree 2aedfd35087c75d17bdbaf4dd56069d44fc75b71
+parent 75158117eb8efe60453f8c077527ac3530c81e38
+author Credit Card Account <Credit Card Account> 1722305889 +0800
+committer \346\244\260\346\235\215
+ <Credit Card Account> 1722305889 +0800
+
+Credit Card Account`
```

Comment:
Thank you for the added test!  It definitely does demonstrate the panic and the fix, which is fantastic.

I spent some time, for my own benefit and future reference, looking into what's going on here.  The code added in commit 930b3ffc6ab4b3ea651a4aa34aaf6c3ed4dc623c of PR #10 added support for multi-line continuations of extra headers, which Git [allows](https://github.com/git/git/blob/406f326d271e0bacecdb00425422c5fa3f314930/commit.c#L1480-L1485) if they start with a leading space character.

However, the code isn't handling the case where an invalid continuation line appears after a regular header.  This can be demonstrated with a test string that is valid except for the extra space before one header line, for instance:
```suggestion
func TestCommitInvalidMultilineNonExtraHeader(t *testing.T) {
	cc := `tree 2aedfd35087c75d17bdbaf4dd56069d44fc75b71
parent 75158117eb8efe60453f8c077527ac3530c81e38
author Jane Doe <jane@example.com> 1503956287 -0400
committer Jane Doe <jane@example.com> 1503956287 -0400
 foo bar

initial commit`

```

You don't need to accept this suggestion, as it doesn't really affect the usefulness of the test, but I wanted to clarify what was going on for my own sake.